blog/2025-12-04-sharded-repodata-improvements.mdx

```diff
@@ -0,0 +1,275 @@
+
+There have been several attempts to address this problem over the past six years, including
+[reducing the size of repodata.json](https://www.anaconda.com/blog/how-we-made-conda-faster-4-7) and
+[incrementally updating repodata by patching](https://conda.org/blog/2023-05-05-how-we-reduced-the-conda-index-fetch-bandwidth/#ship-repodatajlap-incremental-repodata).
+Both of these previous attempts to make repodata fetching more efficient still weren't as performant
+as they could be. This led the engineers at [prefix-dev](https://prefix.dev) who work on the
+[pixi package manager](https://pixi.sh) to innovate and draft [CEP 16](https://conda.org/learn/ceps/cep-0016),
+with the help of the entire conda community, that defines a new mechanism for fetching repodata using a sharded
+approach.
+
+This approach works by splitting up the large repodata into multiple "shards". Each package has its
+own shard which is much smaller than the total repodata. So, when a package is installed, we only
+need to fetch what we need. This results in a much smaller download size and a faster overall experience.
+
+### Bringing sharded repodata to conda and conda-forge
+
+CEP 16 defined a clear way to speed up solves, and we’ve been doing the work needed to make that a reality in conda.
+
+- Earlier this year, we updated [conda-index](https://github.com/conda/conda-index/blob/main/CHANGELOG.md#060-2025-03-27)
+  so channels can generate sharded repodata.
+- Most recently, we added support in [conda-libmamba-solver](https://github.com/conda/conda-libmamba-solver/releases/tag/25.11.0),
+  now in beta, so the conda CLI can consume the new repodata format.
+- [anaconda.org](https://anaconda.org) team at Anaconda worked with the conda-forge community to enable the hosting. We plan to work with other channels to add support as the beta progresses.
+  infrastructure to serve the new sharded format.
+
+With all of this in place, conda-forge is now publishing sharded repodata, and conda will use it automatically
+as soon as you enable the feature.
+
+In the next section, we share the performance improvements we've seen so far.
+
+## Performance improvements
+
+To compare the performance improvements between the sharded and non-sharded approach, we used two different
+environment creation scenarios: **Python** and **Data Science**. The Python scenario just fetches the package
+`python` and the Data Science scenario fetches the packages, `pandas`, `plotly` and `scipy`. We benchmarked
+just the repodata fetching itself (check out the script we used
+[here](https://github.com/travishathaway/perfpy-conda/blob/d749765e1c124fb518784a7a2d91f3748bd32c8c/scripts/sharded_repodata_fetching.py)).
+
+
+:::info
+We used a special, experimental **conda-forge-sharded** channel because at the time of
+profiling, sharded repodata was not available on conda-forge via anaconda.org.
+
+The benchmarks were also run inside a `linux/arm64` Docker container running on an Apple M1 Pro.
```

Comment:
Ah, arm architecture names... It's `osx-arm64` and `linux-aarch64`, even though both run on the M1. I'd be curious what `conda info` in that container gives you, but if you're installing "native" packages from conda-forge, you're getting packages from the `linux-aarch64` subfolder. 

All that to say, it might not matter much to most readers, but I think this is confusing because conda-forge doesn't use `linux/arm64` (note also slash instead of hyphen) anywhere as a name for the target or the subfolder.

PS. Obligatory [Linus rant](https://lkml.org/lkml/2012/7/15/133) on this

Reply:
That's the platform identifier for Docker containers, I think?